src/core/namespace.c

```diff
@@ -287,10 +288,18 @@ static int apply_mount(
 
                 /* First, get rid of everything that is below if there
                  * is anything... Then, overmount it with an
-                 * inaccessible directory. */
+                 * inaccessible path. */
                 umount_recursive(m->path, 0);
 
-                what = "/run/systemd/inaccessible";
+                // mount() always follow symlinks, so we follow symlinks during the stat() as well
```

Comment:
@lucab I suggested to @alepuccetti OOB to document that symlinks are resolved on the host's root directory. I kind of changed my mind: instead of writing complicated code to resolve symlinks, I suggest to document the current behaviour (i.e. ask users to pay attention to how symlinks are resolved, or they can also avoid adding paths with symlinks in {Inaccessible/ReadOnly}Paths).